PYTHON/manifest/manifest.py

```diff
@@ -113,28 +133,63 @@ def func(name: str, input_type: str):
     elif issubclass(return_type, DataContainer):
         create_output("default", return_type.__name__)
     # Multiple outputs
+    elif hasattr(return_type, "__total__") and return_type.__total__:
+        # Get the defined attributes and their type annotations
+        attributes = return_type.__annotations__
+
+        for attribute, annotation in attributes.items():
+            if (not inspect.isclass(annotation) and node_name not in SPECIAL_NODES) or (
+                inspect.isclass(annotation)
+                and not issubclass(annotation, DataContainer)
+            ):
+                raise TypeError(
+                    "Return type must be a DataContainer or a DataClass"
+                    f"consisting of only DataContainers as fields, got {return_type} for {node_name}"
+                )
+            try:
+                output_type = (
+                    union_type_str(annotation)
+                    if is_union(annotation)
+                    else annotation.__name__
+                )
+            except Exception:
+                if node_name in SPECIAL_NODES:
```

Comment:
Should do this check first and return early instead of using exception base control flow